### PR TITLE
Update Windows build instructions for cmake 3.14

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Updates the instructions for building on Windows with CMake 3.14. The platform is no longer a part of the generator name and must be selected separately.

**To test:**

Review the text.

*There is no associated issue.*

*This does not require release notes* because **it is a developer docs change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by dtasev — DISMISSED** [review]
Looks good, instructions work, and the CMake project from 3.14 seems to be working & compiling well.

@martyngigg People upgrading from <3.14 will get an error about `googletest` being built with a different compiler (or something similar). Do you think we should add something about that in the notes? The fix for that is deleting `build/googletest-*`

Note: that happens only if you `Delete Cache` before regenerating the build

File: dev-docs/source/BuildingWithCMake.rst [hunk 1]
```diff
@@ -73,13 +73,13 @@ From the CMake gui
 * Start it and click the "Browse Source" button to point to ``/path/to/Mantid``.
 * Click "Browse Build" and point to the directory you want to build into - it's recommended that you create a new directory for this (see above), though it can be the same as the source directory.
 * Click "Configure" down near the bottom of the window.
-* A new window will appear asking which 'Generator' you want to use.
+* A new window will appear asking which 'Generator' you want to use:
 
   * Linux/Mac developers should choose ``Ninja``
-  * Windows developers should choose ``Visual Studio 15 2017 Win64`` (the Win64 is very important)
+  * Windows developers should choose ``Visual Studio 15 2017`` and in the _Optional platform for generator\_ box select ``x64``. If you see errors related the HDF5 then you have most likely selected the wrong platform.
```
Comment (DanNixon) [inline dev-docs/source/BuildingWithCMake.rst]:
```suggestion
  * Windows developers should choose ``Visual Studio 15 2017`` and in the _Optional platform for generator\_ box select ``x64``. If you see errors related to HDF5 then you have most likely selected the wrong platform. 
```

**Review by DanNixon — CHANGES_REQUESTED** [review]
Small typo.

**Comment by martyngigg** [comment]
Thanks, Dan!

**Review by DanNixon — APPROVED** [review]

**Pull request merged**